client/src/site-search/form.tsx

```diff
@@ -1,149 +1,118 @@
-                updateQuery({ q: query.q, locale: [value], page: "" });
-              }
-            }}
-          >
-            <option value={locale.toLowerCase()}>
+      {locale.toLowerCase() !== "en-us" && (
+        <p className="advanced-option">
+          <b>Language</b>{" "}
+          {!queryLocales.length ||
+          (queryLocales.length === 1 &&
+            equalLocales(queryLocales, [locale])) ? (
+            <i>
               {LANGUAGES.get(locale.toLowerCase())?.native} (
               {LANGUAGES.get(locale.toLowerCase())?.English})
-            </option>
-
-            <option value="en-us">{LANGUAGES.get("en-us")?.native})</option>
-            <option value="both">Both</option>
-          </select>
-        </div>
+            </i>
+          ) : (
+            <Link
+              to={`?${appendURL(searchParams, {
+                locale: [locale],
+                page: undefined,
+              })}`}
+            >
+              {LANGUAGES.get(locale.toLowerCase())?.native} (
+              {LANGUAGES.get(locale.toLowerCase())?.English})
+            </Link>
+          )}
+          {" | "}
+          {queryLocales.length && equalLocales(queryLocales, ["en-us"]) ? (
+            <i>{LANGUAGES.get("en-us")?.native}</i>
+          ) : (
+            <Link
+              to={`?${appendURL(searchParams, {
+                locale: ["en-US"],
+                page: undefined,
+              })}`}
+            >
+              {LANGUAGES.get("en-us")?.native}
+            </Link>
+          )}
+          {" | "}
+          {queryLocales.length === 2 &&
+          equalLocales(queryLocales, [locale, "en-us"]) ? (
+            <i>Both</i>
+          ) : (
+            <Link
+              to={`?${appendURL(searchParams, {
+                locale: [locale, "en-US"],
+                page: undefined,
+              })}`}
+            >
+              Both
+            </Link>
+          )}
+        </p>
       )}
-
-      {/* Rank choice */}
-      <div className="advanced-option">
-        <label htmlFor="id_sort">Sort</label>
-        <select
-          id="id_sort"
-          value={query.sort ? query.sort : "best"}
-          onChange={(event) => {
-            const { value } = event.target;
-            updateQuery({ q: query.q, locale: query.locale, sort: value });
-          }}
-        >
-          <option value="best">Best</option>
-          <option value="relevance">Relevance</option>
-          <option value="popularity">Popularity</option>
-        </select>
-      </div>
     </div>
   );
 }
+
+// Return true if two arrays, independent of case and order are equal.
+// E.g. `['foo', 'Bar']` is equal to `['bar', 'FoO']`
+function equalLocales(list1: string[], list2: string[]) {
+  if (list1.length !== list2.length) {
+    return false;
+  }
+  const list1LC = list1.map((x) => x.toLowerCase());
+  const list2LC = list2.map((x) => x.toLowerCase());
+  return list1LC.every((x) => list2LC.includes(x));
```

Comment:
Hm, I can imagine, this would miss some cases …
Say, `list1 = ['a', 'a', 'b']` and `list2 = ['a', 'b', 'b']`.

Reply:
Those are different lists. One has 2 'a' and the other has 2 'b'.

Reply:
Allow me to say that the best way to use one object that is the same as another is to use the same object